cloud/aws/services/ec2/routetables.go

```diff
@@ -21,10 +21,69 @@ import (
 )
 
 func (s *Service) reconcileRouteTables(in *v1alpha1.Network) error {
-	// TODO(vincepri): add reconcile code when NAT gateways are ready.
+	subnetRouteMap, err := s.describeVpcRouteTablesBySubnet(in.VPC.ID)
+	if err != nil {
+		return err
+	}
+
+	for _, sn := range in.Subnets {
+		if _, ok := subnetRouteMap[sn.ID]; ok {
+			// TODO(vincepri): if the route table ids are both non-empty and they don't match, replace the association.
+			// TODO(vincepri): check that everything is in order, e.g. routes match the subnet type.
+			continue
+		}
+
+		// For each subnet that doesn't have a routing table associated with it,
+		// create a new table with the appropriate default routes and associate it to the subnet.
```

Comment:
How would this work as we extend to multi-az deployments? This comments seems to indicate that we would have a route table per-subnet rather than a route-table per public/private for the VPC.

Reply:
This part should support multi-az deployments just fine I believe. Even with a routing table for each subnet, on creation, we'd pick a NAT gateway to associate using `getNatGatewayForSubnet`, which should follow some rules to pick the best gateway (i.e. same AZ, number of subnets already associated, etc).